src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CQLSessionCache.java

```diff
@@ -103,20 +108,52 @@ private CqlSession getNewSession(SessionCacheKey cacheKey) {
                           host, operationsConfig.databaseConfig().cassandraPort()))
               .collect(Collectors.toList());
 
-      return new TenantAwareCqlSessionBuilder(
-              stargateRequestInfo.getTenantId().orElse(DEFAULT_TENANT))
-          .withLocalDatacenter(operationsConfig.databaseConfig().localDatacenter())
-          .addContactPoints(seeds)
-          .withAuthCredentials(
-              Objects.requireNonNull(databaseConfig.userName()),
-              Objects.requireNonNull(databaseConfig.password()))
-          .build();
+      try {
+        return Uni.createFrom()
+            .completionStage(
+                new TenantAwareCqlSessionBuilder(
+                        stargateRequestInfo.getTenantId().orElse(DEFAULT_TENANT))
+                    .withLocalDatacenter(operationsConfig.databaseConfig().localDatacenter())
+                    .addContactPoints(seeds)
+                    .withAuthCredentials(
+                        Objects.requireNonNull(databaseConfig.userName()),
+                        Objects.requireNonNull(databaseConfig.password()))
+                    .buildAsync())
+            .runSubscriptionOn(Infrastructure.getDefaultWorkerPool())
```

Comment:
Why are we creating a Uni and then do get()? Is it only to make session created on worker thread? IMO this logic may not help since the code calling the cache is blocked (cqlSessionCache.getSession()).